src/server/listener.cc

```diff
@@ -95,6 +95,12 @@ Listener::~Listener() {
         shutdown();
     if (acceptThread.joinable())
         acceptThread.join();
+   
+    if (listen_fd > 0)
```

Comment:
Similar to comment for Epoll class.  Test for ">= 0", and set listen_fd to -1, not 0, after you manually close it.

Reply:
Yap you are right, changes applied